src/wallet/wallet2.cpp

```diff
@@ -3424,6 +3447,44 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
         m_device_name = m_key_device_type == hw::device::device_type::LEDGER ? "Ledger" : "default";
       }
     }
+    
+    if (json.HasMember("original_keys_available"))
+    {
+      GET_FIELD_FROM_JSON_RETURN_ON_ERROR(json, original_keys_available, int, Int, false, false);
+      m_original_keys_available = field_original_keys_available;
+      if (m_original_keys_available)
+      {
+	GET_FIELD_FROM_JSON_RETURN_ON_ERROR(json, original_address, std::string, String, false, std::string());
+	if (!field_original_address_found)
+	{
+	  LOG_ERROR("Field original_address not found in JSON");
+	  return false;
+	}
+	address_parse_info info;
+	bool ok = get_account_address_from_str(info, m_nettype, field_original_address);
+        if (!ok)
+	{
+	  LOG_ERROR("Failed to parse original_address from JSON");
+	  return false;
+	}
+	m_original_address = info.address;
+	GET_FIELD_FROM_JSON_RETURN_ON_ERROR(json, original_view_secret_key, std::string, String, false, std::string());
+	if (!field_original_view_secret_key_found)
+	{
+	  LOG_ERROR("Field original_view_secret_key not found in JSON");
+	  return false;
+	}
+	ok = epee::string_tools::hex_to_pod(field_original_view_secret_key, m_original_view_secret_key);
+	if (!ok)
+	{
+	  LOG_ERROR("Failed to parse original_view_secret_key from JSON");
```

Comment:
return false ?

Reply:
Yes, clear oversight, good you catched that. Added.